### DISCUSSION TITLE
App Store Release?

### DISCUSSION THREAD

**Discussion by cryptoAlgorithm** [cryptoAlgorithm/Swiftcord#17] [Ideas]
So it turns out 3rd party clients _might_ not be against Discord's ToS after all. Someone suggested releasing this on the App Store. If I release Swiftcord on the App Store, would you download it? How about if it was paid? Do you foresee any with putting Swiftcord on the App Store?

We should also probably continue the discussion from #15 here.

**Comment by rrroyal** [comment]
I'm pretty sure 3rd party clients aren't against Discord's ToS, **but submitting it to App Store might be** (you need to have Discord's permission to use their API if you were to submit Swiftcord to App Store iirc). It would also be worse if Discord changed their API and you'd need to deploy a hotfix - app could be stuck in review for weeks.

Distributing the app with Sparkle (or using Homebrew) could be better 😁

**Reply by cryptoAlgorithm** [reply]
Yup, that is one of my concerns. Why would I need Discord's permission to use their API though? And yes, app review is extremely annoying, from a fellow developer's experience xD

**Reply by cryptoAlgorithm** [reply]
A developer account could still allow things like autoupdate (with sparkle, in the future) and the first time the app is opened to proceed more smoothly, as I'd be able to sign it so macOS "trust" it.

**Reply by rrroyal** [reply]
> Yup, that is one of my concerns. Why would I need Discord's permission to use their API though? And yes, app review is extremely annoying, from a fellow developer's experience xD

App Store ToS I'm pretty sure; even if not, they could just take the app down (like [Leah's](https://twitter.com/LeahLundqvist) Discord watchOS app) :/

**Reply by cryptoAlgorithm** [reply]
Yeah, but I'm sure apple doesn't have anything against native Discord clients, _right_? If I needed to ask the API writer for permission before releasing an app that used it, half the apps wont exist on the App Store today.

**Reply by rrroyal** [reply]
> Yeah, but I'm sure apple doesn't have anything against native Discord clients, right?

yup, they don't care

> If I needed to ask the API writer for permission before releasing an app that used it, half the apps wont exist on the App Store today.

Apps that use public api's are fine - it's a whole different story with private ones (i.e. Discord, Instagram, etc.)

**Reply by cryptoAlgorithm** [reply]
Ah I see. It could be argued that the Discord API _is_ documented, since as of now only a few APIs used by Swiftcord aren't documented on the [official developer docs](https://discord.com/developers/docs/)

**Reply by rrroyal** [reply]
I think the biggest thing they would argue is actual authentication - logging in with user token _isn't_ documented, nor public :/

**Reply by cryptoAlgorithm** [reply]
It is :), all bots use the token to login, and I did not reverse-engineer their email+pw/qr login since I just load their page in a WebView and pull the token once its available

**Reply by rrroyal** [reply]
yup, I'm doing the same, the thing is, you theoretically need to send `intents` and `properties` in `IDENTIFY`, but you're not doing it (because you're acting as a user, not a bot) - they would probably catch you by that.

i could be totally wrong though ¯\\\_(ツ)_/¯

**Reply by cryptoAlgorithm** [reply]
Actually ~~by reverse engineering~~ I realised Discord's own client also sends the token in identify and authenticates using that, so what I'm doing is no different from what the official client does lol

**Reply by cryptoAlgorithm** [reply]
Also, since this discussion is probably gonna get long, we should carry on in the [Swiftcord Discord server](https://discord.gg/he7n6MGDXS) so I know when you've sent another message, since GitHub for some reason didn't show that reply until I reloaded the page

**Reply by Candygoblen123** [reply]
I'd like to add that apple wouldn't allow this on the App Store.  I've had an app removed before because I used the public Youtube API, and we had to attach  the Youtube Developer agreement.

**Reply by cryptoAlgorithm** [reply]
Oh that is another bummer. I think I'll just put this on TestFlight if I get a dev account xD

**Comment by jasperweiss** [comment]
Note that you don't _need_ to release it in the app store in order for it to run without warnings. You only need to get the app notarised which is just an automated check for malicious behaviour, entitlements, use of the hardened runtime among several other things but nothing related to licensing or ToS AFAIK. See [Apple's documentation](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution).

**Reply by cryptoAlgorithm** [reply]
Yes, actually a kind contributor, @Candygoblen123, has helped set up a GitHub action that builds every commit and notarises it with their own dev account. Really appreciate that, compiled builds can be found in the artefacts of the action run.

**Comment by MrPaschenko** [comment]
![Bonnet_macaque_(Macaca_radiata)_Photograph_By_Shantanu_Kuveskar](https://github.com/user-attachments/assets/7eeab01e-6747-4d8f-b902-6f96799cdbb2)
